pkg/clients/mysql/mysql.go

```diff
@@ -21,24 +21,36 @@ type mySQLDB struct {
 	dsn      string
 	endpoint string
 	port     string
+	tls	     string
 }
 
 // New returns a new MySQL database client.
-func New(creds map[string][]byte) xsql.DB {
+func New(creds map[string][]byte, tls string) xsql.DB {
 	// TODO(negz): Support alternative connection secret formats?
 	endpoint := string(creds[xpv1.ResourceCredentialsSecretEndpointKey])
 	port := string(creds[xpv1.ResourceCredentialsSecretPortKey])
+	username := string(creds[xpv1.ResourceCredentialsSecretUserKey])
+	password := string(creds[xpv1.ResourceCredentialsSecretPasswordKey])
+	dsn := DSN(username, password, endpoint, port, tls)
+
 	return mySQLDB{
-		dsn: fmt.Sprintf("%s:%s@tcp(%s:%s)/",
-			creds[xpv1.ResourceCredentialsSecretUserKey],
-			creds[xpv1.ResourceCredentialsSecretPasswordKey],
-			endpoint,
-			port),
+		dsn:      dsn,
 		endpoint: endpoint,
 		port:     port,
+		tls:      tls,
 	}
 }
 
+// DSN returns the DSN URL
+func DSN(username, password, endpoint, port, tls string) string {
```

Comment:
As you are touching this, would you consider improving it a little bit? With PostgreSQL we noticed the credentials weren't properly escaped, this was fixed in https://github.com/crossplane-contrib/provider-sql/pull/62 and should probably also be implemented for MySQL (issue https://github.com/crossplane-contrib/provider-sql/issues/70)

No prob if you don't want to, we can pick it up in another PR.

Reply:
I think I have this updated for the escaping.